src/python/pants/pantsd/pants_daemon.py

```diff
@@ -195,4 +195,7 @@ def pre_fork(self):
 
   def post_fork_child(self):
     """Post-fork() child callback for ProcessManager.daemonize()."""
+    for service in self._services:
+      service.post_fork()
```

Comment:
rather than plumbing a `post_fork` hook to all services and setting up the handler in the `SchedulerService`'s `post_fork`, you might be better off just calling the panic handler installer directly - at exactly this location?

in `PantsDaemonLauncher` (a `Subsystem` that sets up pantsd) you can declare a dependency on the `Native` subsystem to get a handle to it - and then pass that directly into the `PantsDaemon` constructor for access to e.g. `Native.global_instance().set_panic_handler()`.